packages/flame/lib/src/components/component.dart

```diff
@@ -326,6 +353,57 @@ class Component {
     _children?.removeAll(cs);
   }
 
+  @internal
+  void mount() {
+    assert(_parent != null);
+    assert(_loaded && !_mounted && !_prepared);
+    if (_parent!.isMounted) {
+      onMount();
+      _prepared = true;
```

Comment:
What is the difference between `_mounted` and `_prepared` now?

Reply:
`_prepared` is set to true after `onMount` runs (I know, sounds weird). Then, at some later time, the component is picked up from the "children queue" and added to the `ComponentSet` of its parent. At that moment, when the component is finally added to its parent's children, the component is considered fully mounted and is marked with `_mounted = true`. 

Immediately after that we perform `remount()` on all children of the component (if any), in order to execute `onGameResize/onMount` for them. By the time this method finishes, both the original component, and all its children will be fully mounted.

Reply:
Right, and in what situations do we need to check `_prepared`?

Reply:
That queue is being traversed when the parent is being mounted right? Couldn't `mount` run right before it is being added there instead and then the `_parent!.isMounted` check and the `_prepared` variable wouldn't be needed here?

Reply:
You're right, this does make the code simpler.
The only trouble was that when a component starts adding new components in its `onMount`, and it modifies the `childrenQueue` while it is being iterated -- however, I added a check to guard against that.